### ISSUE TITLE
Erratic phrase search

### ISSUE THREAD

**Issue by trailerparkdev** [meilisearch/meilisearch#2257]
Hi MeiliSearch and thanks for the good work.

I have a strange bug on phrase search.

Consider the following curl : 
`curl -X POST 'http://172.19.5.3:7700/indexes/elements/search' -H 'Content-Type: application/json' --data-binary '{ "q": "\"temp travailleur domicil\"" }' | jq`

I get results: 
```
{
  "hits": [
    {
      "tokenized": "mis oeuvr disposit présenc cod effect entrepris calcul conform disposit 1 ° salair titulair contrat travail dur indétermin temp travailleur domicil préhens intégral compt effect entrepris 2 ° salair titulair contrat travail dur détermin",
      "uid": "c20700255ba6ed665712fa318dbe20f337d43499debbf1e9acf165be517f4047_25"
    }
  }
  "nbHits": 1,
  "exhaustiveNbHits": false,
  "query": "\"temp travailleur domicil\"",
  "limit": 20,
  "offset": 0,
  "processingTimeMs": 1
}
```

Now a phrase search with other terms **that are in the result above in this exact order:** 
`curl -X POST 'http://172.19.5.3:7700/indexes/elements/search' -H 'Content-Type: application/json' --data-binary '{ "q": "\"salair titulair\"" }' | jq`

```
{
  "hits": [],
  "nbHits": 0,
  "exhaustiveNbHits": false,
  "query": "\"salair titulair\"",
  "limit": 20,
  "offset": 0,
  "processingTimeMs": 1
}
```

This stange behavior happens on a dockerized Meili 0.24. 

So my questions are:

1: any idea why?

2: we use our own tokenizers (as you see the result document is a string of stemmed tokens) for reasons that would be too long to develop here. Is the fact that we only index strings of stemmed tokens a problem?

3: if so, could you provide a setting where Meili's tokenizer could be disabled when users **need** to use their custom ones? I believe that would open Meili to a much wider range of use cases. In our context, we're investigating a switch from Elastic to Meili. If we can't use our own tokenizers (they're multilingual and the result of a lot of work), it will be a no go. And I'm inclined to believe that many pro shops are in this same case.

Thanks for the time you'll take to reply. French devs rule!

**Comment by ManyTheFish** [comment]
Hey @trailerparkdev!

I see that your Meilisearch version is 0.24 and we fixed a similar issue in [0.25](https://github.com/meilisearch/meilisearch/releases/tag/v0.25.0), could you retry this with the [last version of Meilisearch](https://github.com/meilisearch/meilisearch/releases/tag/v0.26.1)?

thanks!

**Comment by trailerparkdev** [comment]
Salut @curquiza.

Thanks for your quick reply. Switching to 0.26 gives the expected results. 

Would you confirm that tokenization on your side is necessary even though we do ours and only index tokenized strings on searchable fields? 

And, provided it's not necessary, would you consider exposing an option to disable it ?

Have a nice day.

**Comment by curquiza** [comment]
Hello @trailerparkdev 

Yes, our tokenization on our side is a mandatory step currently, you cannot disable it.

However, I would really know better about your use case. If you have the time to open a discussion in [this repository](https://github.com/meilisearch/product/discussions) it would definitely help our product team 🙂

Glad to see your issue is fixed with v0.26.0 👍 

I close this issue since it's fixed!
